### PR TITLE
Add entry for form-associated custom element

### LINKED ISSUE #203: Add form-associated custom element
Mappings for form-associated custom element:

No implied/default role.

Roles that may be used: application, button, checkbox, combobox, group, listbox, progressbar, radio, searchbox, slider, spinbutton, textbox

Global and role-appropriate aria attributes.

----

Notes:
- `application` was included because it could represent a new type of form control
- `group` was included because it's the default role for `<fieldset>` (although `group` can also represent non-form-associated groups as well)
- listbox was included because of `<select>` multiple or size > 1
- html output is a form control that maps to role=status. This wasn't included because it's not clear whether allowing role=status (without other live region roles) is necessary.

### PR DESCRIPTION
**Pull request by carmacleod**
Closes #203 

- Add entry for form-associated custom element
- Prefix autonomous custom element id with 'el-' for consistency with other elements and with html-aam.

***
<a href="https://pr-preview.s3.amazonaws.com/carmacleod/html-aria/pull/204.html" title="Last updated on Dec 14, 2019, 8:50 PM UTC (a2bb14d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aria/204/bb66bc6...carmacleod:a2bb14d.html" title="Last updated on Dec 14, 2019, 8:50 PM UTC (a2bb14d)">Diff</a>

### CODE REVIEW

**Comment by carmacleod** [comment]
I think maybe respec doesn't like the dash in `<a>form-associated custom elements</a>`?
https://respec.org/xref/?term=form-associated+custom+element&specs=html&types=_CONCEPT_

But even if I replace the `-` with a space, there still seems to be a `_CONCEPT_` error?
https://respec.org/xref/?term=form+associated+custom+element&specs=html&types=_CONCEPT_

The concept link in the html spec is: https://html.spec.whatwg.org/multipage/custom-elements.html#form-associated-custom-element
and its definition (in the DOM) looks like this:
```
<dfn id="form-associated-custom-element">form-associated custom element</dfn>
```

@marcoscaceres Is there some kind of `data-lt` magic I need to do over on the html side to get this html-aria pr to build? Or can it be fixed in this pr somehow? Sorry - I'm very new to respec. :)

**Comment by carmacleod** [comment]
So, I think I've figured it out (the presence or absence of the dash was a red herring).
I think the definition of "form-associated custom element" in the html spec source needs `data-export`:
```
<dfn data-export="">form-associated custom element</dfn>
```

Pretty sure that is what is making "autonomous custom element" link up correctly. That source is:
```
<dfn data-export="">autonomous custom element</dfn>
```

Is there a way to test this without first merging the `data-export` code into the html spec?

**Comment by carmacleod** [comment]
This should build ok now, I hope, because this pr has been merged: https://github.com/whatwg/html/pull/5153

**Comment by scottaohara** [comment]
Marked as non substantive for IPR from ash-nazg.

**Review by scottaohara — COMMENTED** [review]
since we spoke about this offline, i'm tentatively ok with this PR, pending any additional feedback that @stevefaulkner has to provide.

also still got the build failure to contend with due to form associated custom elements not being recognized yet per your whatwg PR

**Comment by carmacleod** [comment]
@marcoscaceres I tried to fix this "missing dfn" respec error, but the build is still failing, so I'm stuck now and need help please.

**Comment by carmacleod** [comment]
@marcoscaceres ReSpec is happy now, and the build is working! Apparently the database that ReSpec uses to look up the definitions can take a while to update. I rebuilt just now (17 hours later), and the dfn is no longer missing. Thanks to @annevk for the suggestion!

@scottaohara @stevefaulkner This PR is now ready for review.

File: index.html [hunk 1]
```diff
@@ -919,6 +919,36 @@ <h2 id="docconformance">
...
+                <a href="#index-aria-button">`button`</a>,
+                <a href="#index-aria-checkbox">`checkbox`</a>,
+                <a href="#index-aria-combobox">`combobox`</a>,
+                <a href="#index-aria-group">`group`</a>,
+                <a href="#index-aria-listbox">`listbox`</a>,
+                <a href="#index-aria-progressbar">`progressbar`</a>,
+                <a href="#index-aria-radio">`radio`</a>,
+                <a href="#index-aria-searchbox">`searchbox`</a>,
+                <a href="#index-aria-slider">`slider`</a>,
+                <a href="#index-aria-spinbutton">`spinbutton`</a>
```
Comment (scottaohara) [inline index.html]: we need `switch` as well.

Comment (carmacleod) [reply index.html]: Good catch. Added.

**Comment by carmacleod** [comment]
I think we need `radiogroup` also?

**Comment by scottaohara** [comment]
>I think we need radiogroup also?

yes, agreed.

**Review by scottaohara — APPROVED** [review]
I think this makes sense.

**Comment by marcoscaceres** [comment]
Yes, sometimes it can take 24 hours to get an update... usually it happens pretty quick :)

**Pull request merged**